terragraph_planner/los/elevation.py

```diff
@@ -286,3 +312,66 @@ def __sub__(self, other_geogrids: "Elevation") -> "Elevation":
             raise DataException(
                 "Subtraction operation is only supported on GeoGrids with same metadata"
             )
+
+
+class ElevationKDTree:
+    """
+    A wrapper around a KDTree containing elevation data (x, y, altitude) derived
+    from surface elevation data, using a dict of dicts as the storage mechanism.
+    The default resolution of the points is 1m.
+    """
+
+    def __init__(
+        self, elevation_data: List[Tuple[float, float, float]]
+    ) -> None:
+        # Ensure elevation_data is not empty
+        if len(elevation_data) == 0:
+            elevation_data = [(0, 0, 0)]
```

Comment:
Not sure whether using (0,0,0) as default is good. Ideally, we just initialize some empty placeholders here and handle the special case in the following function based on the placeholders.

Reply:
The length should be always positive. I added a check at the end of `Elevation.get_data_as_list()`, which is the source of the `elevation_data` in this constructor, to make sure its length is valid.